### PR TITLE
ringct: pseudoOuts moved to prunable in the simple bulletproof case

### PR DESCRIPTION
**Pull request by moneromooo-monero**
Saves 64 bytes non prunable data per typical tx

This breaks v7 consensus, will require a testnet reorg from v6

### CODE REVIEW

File: src/ringct/rctSigs.cpp [hunk 1]
```diff
@@ -337,7 +361,8 @@ namespace rct {
 
       std::stringstream ss;
       binary_archive<true> ba(ss);
-      const size_t inputs = rv.pseudoOuts.size();
+      CHECK_AND_ASSERT_THROW_MES(!rv.mixRing.empty(), "Empty mixRing");
+      const size_t inputs = is_simple(rv.type) ? is_bulletproof(rv.type) ? rv.p.pseudoOuts.size() : rv.pseudoOuts.size() : rv.mixRing[0].size();
```
Comment (stoffu) [inline src/ringct/rctSigs.cpp]: Is this change necessary at all? The variable `inputs` is only relevant for serializing pre-Bulletproof pseudoOuts, so the existing code `inputs = rv.pseudoOuts.size();` seems to just work (i.e. with Bulletproof+Simple, it's ignored in `serialize_rctsig_base` anyway, and with pre-BP/BP+Full, it won't matter either).

Comment (moneromooo-monero) [reply src/ringct/rctSigs.cpp]: Yes, because relying on the knowledge that the other piece of code won't actually use the parameter in some subset of cases is brittle.

Comment (stoffu) [reply src/ringct/rctSigs.cpp]: OK.

Comment (luigi1111) [reply src/ringct/rctSigs.cpp]: Using pseudoOuts at all feels wrong somehow. Couldn't you just check is_simple and use mixRing.size()/mixRing[0].size()?

Comment (moneromooo-monero) [reply src/ringct/rctSigs.cpp]: True, this would simplify a bit, I'll change it.
